### PR TITLE
Change link order for static links

### PR DESCRIPTION
**Pull request by seelabs**
zlib needs to be later in the link list when linking statically on Ubuntu 12.04. Static linking used to work on Ubuntu 12.04. I have not investigated what specific change caused the failure.

I tested this change on Ubuntu 12.04, 14.04, 15.10 and Fedora 22 or both static and dynamic linking.

### CODE REVIEW

**Comment by ximinez** [comment]
Reviewers: @ximinez @miguelportilla

File: SConstruct [hunk 1]
```diff
@@ -393,16 +393,20 @@ def config_base(env):
...
     if not 'HASSTATICLIBS' in env:
         env['HASSTATICLIBS'] = True
         env['STATICLIBS'] = ''
-        env.Replace(LINKCOM=env['LINKCOM'] + " -Wl,-Bstatic $STATICLIBS")
-    c = env['STATICLIBS']
-    if c == None: c = ''
-    for f in libs:
-        c += ' -l' + f 
-    env['STATICLIBS'] = c
+        env['DYNAMICLIBS'] = ''
```
Comment (ximinez) [inline SConstruct]: Why bother initializing `env['STATICLIBS'] and env['DYNAMICLIBS']` here when you're about to initialize them on line 406 anyway? Or do you explicitly want to make sure they're _empty_ in this case?

**Comment by ximinez** [comment]
Looks good. Default builds work in Windows ("Static linking is only implemented for linux."). Default and static builds work in Linux. (I used `./Builds/Test.py -k -v -- -j 8  debug && rm -rf build/ && ./Builds/Test.py -k -v -- -j 8  debug  --static` to test.)

File: SConstruct (hunk 1 above)
Comment (seelabs) [inline SConstruct]: I append to those variables, and they need to be initialized as strings. Line 405 gets the old value, line 409 assigns the new value. Both variables will be empty at line 399, but I can add a check for that assumption.

Comment (seelabs) [inline SConstruct]: I missed line 406, which I think is what you are referring to. You're right, I can re-write this in a simpler way. Will do.

**Comment by seelabs** [comment]
@ximinez Push a commit (hopefully) addressing your comment on `add_static_libs`

**Comment by miguelportilla** [comment]
Tested last commit. All default builds work in Windows and OS X. Default and static builds work in Linux 15.10 (./Builds/Test.py -k -v -- -j8 --static).

**Comment by ximinez** [comment]
:+1:

**Comment by miguelportilla** [comment]
:+1:

**Pull request closed without merging**